tests/python/relax/test_op_legalizer.py

```diff
@@ -0,0 +1,2582 @@
+
+    @I.ir_module
+    class Expected:
+        @R.function
+        def main(x: R.Tensor((2, 64, 7, 7), "float32")) -> R.Tensor(None, "float32", ndim=4):
+            gv = R.call_tir(adaptive_pool, (x,), (2, 64, 1, 1), dtype="float32")
+            return gv
+
+        @T.prim_func
+        def adaptive_pool(
+            rxplaceholder: T.Buffer[(T.int64(2), T.int64(64), T.int64(7), T.int64(7)), "float32"],
+            adaptive_pool_avg: T.Buffer[
+                (T.int64(2), T.int64(64), T.int64(1), T.int64(1)), "float32"
+            ],
+        ):
+            T.func_attr({"tir.noalias": True})
+            adaptive_pool_sum = T.alloc_buffer(
+                [T.int64(2), T.int64(64), T.int64(1), T.int64(1)], dtype="float32"
+            )
+            for i0, i1, i2, i3, i4, i5 in T.grid(
+                T.int64(2), T.int64(64), T.int64(1), T.int64(1), T.int64(7), T.int64(7)
+            ):
+                with T.block("adaptive_pool_sum"):
+                    ax0, ax1, ax2, ax3, rv0, rv1 = T.axis.remap("SSSSRR", [i0, i1, i2, i3, i4, i5])
+                    T.reads(rxplaceholder[ax0, ax1, ax2 * T.int64(7) + rv0, ax3 * T.int64(7) + rv1])
+                    T.writes(adaptive_pool_sum[ax0, ax1, ax2, ax3])
+                    with T.init():
+                        adaptive_pool_sum[ax0, ax1, ax2, ax3] = T.float32(0)
+                    adaptive_pool_sum[ax0, ax1, ax2, ax3] = (
+                        adaptive_pool_sum[ax0, ax1, ax2, ax3]
+                        + rxplaceholder[ax0, ax1, ax2 * T.int64(7) + rv0, ax3 * T.int64(7) + rv1]
+                    )
+            for i0, i1, i2, i3 in T.grid(T.int64(2), T.int64(64), T.int64(1), T.int64(1)):
+                with T.block("adaptive_pool_avg"):
+                    ax0, ax1, ax2, ax3 = T.axis.remap("SSSS", [i0, i1, i2, i3])
+                    T.reads(adaptive_pool_sum[ax0, ax1, ax2, ax3])
+                    T.writes(adaptive_pool_avg[ax0, ax1, ax2, ax3])
+                    T.block_attr({"schedule_rule": "meta_schedule.adaptive_pool_avg"})
+                    adaptive_pool_avg[ax0, ax1, ax2, ax3] = adaptive_pool_sum[
+                        ax0, ax1, ax2, ax3
+                    ] * T.float32(0.020408163265306121)
+
+    mod = OperatorLegalizer(AdaptiveAvgPool2D).transform()
+    tvm.ir.assert_structural_equal(mod, Expected)
+
+
+def test_sum():
+    @I.ir_module
+    class Sum:
+        @R.function
+        def main(x: R.Tensor((1, 2, 3, 4), "float32")) -> R.Tensor(None, "float32", ndim=2):
+            gv: R.Tensor((1, 3), "float32") = R.sum(x, axis=[1, 3])
+            return gv
+
+    @I.ir_module
+    class Expected:
+        @R.function
+        def main(x: R.Tensor((1, 2, 3, 4), "float32")) -> R.Tensor(None, "float32", ndim=2):
+            gv = R.call_tir(sum, (x,), (1, 3), dtype="float32")
+            return gv
+
+        @T.prim_func
+        def sum(
+            rxplaceholder: T.Buffer[(T.int64(1), T.int64(2), T.int64(3), T.int64(4)), "float32"],
```

Comment:
Perhaps a good motivational pt for meta-programming later  cc @MasterJH5574 @Hzfengsy